x/checkpointing/proposal.go

```diff
@@ -30,13 +34,19 @@ func NewProposalHandler(
 	logger log.Logger,
 	ckptKeeper CheckpointingKeeper,
 	mp mempool.Mempool,
-	txVerifier baseapp.ProposalTxVerifier,
+	bApp *baseapp.BaseApp,
+	encCfg *appparams.EncodingConfig,
 ) *ProposalHandler {
-	defaultHandler := baseapp.NewDefaultProposalHandler(mp, txVerifier)
+	defaultHandler := baseapp.NewDefaultProposalHandler(mp, bApp)
+	ckpttypes.RegisterInterfaces(encCfg.InterfaceRegistry)
+
 	return &ProposalHandler{
 		logger:                        logger,
 		ckptKeeper:                    ckptKeeper,
-		txVerifier:                    txVerifier,
+		bApp:                          bApp,
+		txEncoder:                     encCfg.TxConfig.TxEncoder(),
+		txDecoder:                     encCfg.TxConfig.TxDecoder(),
+		txFactory:                     tx.Factory{}.WithChainID(bApp.ChainID()).WithTxConfig(encCfg.TxConfig),
```

Comment:
wdyt about using:
```
encCfg.TxConfig.NewTxBuilder()
```
directly to build transactions instead of ` tx.Factory{}` ?

My main fear is that `txFactory.BuildUnsignedTx` have a lot of internal logic itself, and this logic may change in time (with patch or minor version) and we do not catch this change and hit some consensus error. Ideally we would build this transaction with least amout of logic possible i.e just wrap our message with transaction proto.